design/Alameda/design.md

```diff
@@ -13,26 +13,50 @@ The main purpose of Alameda is to provide prediction and planning to Rook. Rook
     Based on a disk's S.M.A.R.T. value, Alameda predicts how bad a commodity disk is going to fail in a near future. Rook can stop provisioning volumes from a critical status disk.
 - Performance prediction  
-    Alameda learns patterns from the historical performance metrics of persistent volumes and pools. With this knowledge, 
-    for basic features:
-    - Prediction-based Pod (CPU and memory) requests and limits configuration
-    
-This approach has the following benefits:
-    - have more efficient pod resource configuration planning
-    - have more efficient pod allocation planning
-    - provision volumes from low loading pool
-    - recommend disk replacement time
+    Alameda learns patterns from the historical performance metrics of persistent volumes and pools. This approach has the following benefits:
+    - prediction-based pod (CPU and memory) requests and limits configuration
+    - prediction-based pod replication
+    - provision volumes from predicted low loading pool
+    - recommend predicted low loading time window for disk replacement
 - Capacity trending prediction  
     Alameda can also provide capacity usage trending prediction for Rook's storage providers as well as storages exposed by Rook. Rook knows better **when** to add more disks or more nodes for more capacity.
 
 ## How Alameda works
 
-1. Alameda data collector gets metrics from Prometheus (e.g. CPU, memory, Ceph metrics). No Alameda Agent is needed.
+1. Alameda data collector gets metrics from Prometheus (e.g. CPU, memory, Ceph metrics)  
+No Alameda Agent is needed.
 2. Alameda AI engine generates resource prediction
-3. Alameda update cluster CRD: (1) update CRD spec, or (2) update CRD spec with new definitinos of planning
-    No matter they are (1) or (2), Rook needs to change the logic of watching CRD
-3. Alameda resource operator monitors Rook cluster CRD
+3. Alameda resource operator monitors Rook cluster CRD  
+Alameda will monitor rook CRDs with Alameda annotations. For example, Rook user can add ```container.ai/autoscale``` and ```container.ai/diskFailurePrediction``` annotations in their *cluster.yaml* as:
+<pre>
+    apiVersion: v1
+    kind: Namespace
+    metadata:
+      name: rook
+    ---
+    apiVersion: rook.io/v1alpha1
+    kind: Cluster
+    metadata:
+      name: rook
+      namespace: rook
+      <b>annocations:
+        container.ai/autoscale: true
+        container.ai/diskFailurePrediction: true</b>
+    spec:
+      versionTag: v0.5.1
+      dataDirHostPath:
+      storage:
+        useAllNodes: true
+        useAllDevices: false
+        storeConfig:
+          storeType: filestore
+          databaseSizeMB: 1024
+          journalSizeMB: 1024
+</pre>
+
 4. Alameda generates resource operation planning for Rook cluster
+5. Alameda update cluster CRD  
+Alameda will (1) update CRD spec, or (2) update CRD spec with new definitions of planning. No matter they are (1) or (2), Rook needs to change the logic of watching CRD.
 
```

Comment:
Travisn asked:
What is the output of the resource operation planning? How could the rook operator watch for the planning output?

The output of resource operation will be cpu/memory requests and limits. And k8s will update pods with new configuration. Does Rook  operator need to do any response?